src/DotLiquid.Tests/ContextTests.cs

```diff
@@ -531,14 +536,22 @@ public void TestAccessHashesWithHashAccessVariables()
         }
 
         [Test]
+        // Updated test cases to match Ruby and allow a label for an array, such as "array['first']".
+        // See https://github.com/Shopify/liquid/blob/50d1a2ffc91e69c26af2eea278cbc9c4930126a9/test/integration/drop_test.rb#L232
         public void TestHashNotationOnlyForHashAccess()
         {
             _context["array"] = new[] { 1, 2, 3, 4, 5 };
-            _context["hash"] = new { first = "Hello" };
+            _context["hash"] = new { last = "goodbye", first = "Hello" };
+            _context["emptyArray"] = new int[0];
 
             Assert.AreEqual(1, _context["array.first"]);
-            Assert.AreEqual(null, _context["array['first']"]);

```

Comment:
See my other comments this behavior should not be changed. Is there a spec or test case in Liquid that says this is valid syntax?

Reply:
Does this [test](https://github.com/Shopify/liquid/blob/50d1a2ffc91e69c26af2eea278cbc9c4930126a9/test/integration/drop_test.rb#L232.) mean array["first"] is supported by Ruby?

Reply:
Seems so. I will confirm on an actual ruby install, but in the interim please note this change in the description of issues fixed and mark this resolved.

Reply:
See test_hash_notation_only_for_hash_access - https://github.com/Shopify/liquid/blob/master/test/integration/context_test.rb this is not valid after all. I'm not sure what that other test is, but a quick check against my ruby install confirms array.first != array["first"]

Reply:
Reverted.